iris-mpc-cpu/src/protocol/ops.rs

```diff
@@ -265,12 +270,11 @@ pub async fn galois_ring_is_match(
 }
 
 /// Checks that the given dot product is zero.
-pub async fn is_dot_zero(
+pub async fn is_distance_zero(
```

Comment:
I also think the current construction is kind of leaking the locality of the current non-networked implementation.
I would rather, like we did for our inititial CPU iris prototypes have a protocol that is generic over the networking, where the networking can then just be handled by channels as it is now for the local variant and by e.g. GRPC for the networked variant, but have the remaining protocol be the same. This would then mean that we do 3 instances of the local protocol for local testing in 3 different threads, rather than the current protocol having one instance and encapsulating this internally. IMO this is cleaner as otherwise some of the internal functionality is duplicated between these two versions?

Reply:
Agreed. The name and description are updated.